Common/src/Storage/TaskOptions.cs

```diff
@@ -100,3 +75,42 @@ public record TaskOptions(IDictionary<string, string> Options,
                              : defaultOption.EngineType);
   }
 }
+
+public static class GrpcTaskOptionsExt
+{
+  public static TaskOptions ToTaskOptions(this Api.gRPC.V1.TaskOptions taskOption)
+    => new(taskOption.Options,
+           taskOption.MaxDuration.ToTimeSpan(),
+           taskOption.MaxRetries,
+           taskOption.Priority,
+           taskOption.PartitionId,
+           taskOption.ApplicationName,
+           taskOption.ApplicationVersion,
+           taskOption.ApplicationNamespace,
+           taskOption.ApplicationService,
+           taskOption.EngineType);
+
+  public static TaskOptions? ToNullableTaskOptions(this Api.gRPC.V1.TaskOptions? taskOption)
+    => taskOption?.ToTaskOptions();
+}
+
+public static class TaskOptionsExt
+{
+  public static Api.gRPC.V1.TaskOptions ToTaskOptions(this TaskOptions taskOption)
```

Comment:
and break nullable. It would generate a lot of warnings.

And there is a great issue I encountered:
- CreateTask takes an option and an `IAsyncEnumerable<TaskRequest>`
- the conversion from gprc options to internal options does not work because the grpc options is null. It comes from the fact that the implicit conversion does not work well with null (we cannot easily detect that the object is null before the implicit conversion and avoid making this conversion that does not work. Doing so takes a lot of lines of code that are not really self-explanatory)
- so implicit conversion does not work but we still enters task creation through the IAsyncEnumerable
- request are not consummed and application wait
- task creation handle indefinitely.

Reply:
I think you missed my point here. I just suggested to rename the method from `ToTaskOptions` into `ToGrpcTaskOptions`. I said nothing about nullable. The question mark to ask a question, not making the call nullable.

Reply:
oh right ! I will change the name to make it clearer !